### ISSUE TITLE
`std::tests::test_dirs::test_dirs_command` fails on MacOS

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#8528]
### Describe the bug

the tests of #8525 do not pass.

according to the investigations of @fdncred in the comments of the PR
> This is what's going on.
> 
> ```
> > assert ($base_path == $env.DIRS_LIST.0) "list is just pwd after initialization"
> Error:
>   × list is just pwd after initialization
>    ╭─[entry #76:1:1]
>  1 │ assert ($base_path == $env.DIRS_LIST.0) "list is just pwd after initialization"
>    ·        ────────────────┬───────────────
>    ·                        ╰── It is not true.
>    ╰────
> ```
> 
> ```
> > $base_path
> /var/folders/m1/x0gp9jy51s146ttxtz7jlpxh0000gn/T/test_dirs_14f52f6b-23fd-4324-a29e-b7f7f0ca314d
> > $env.DIRS_LIST.0
> /private/var/folders/m1/x0gp9jy51s146ttxtz7jlpxh0000gn/T/test_dirs_14f52f6b-23fd-4324-a29e-b7f7f0ca314d
> ```
> 
> /tmp is a symlink
> ```
> ^ls -al /tmp
> lrwxr-xr-x 1 root wheel 11 Feb  9 03:39 /tmp -> private/tmp
> ```

### How to reproduce

1. have a Mac ( :eyes: )
2. run
```bash
let base_path = (($nu.temp-path) | path join $"test_dirs_(random uuid)" | path expand )
mkdir $base_path
cd $base_path

use /path/to/crates/nu-utils/standard_library/std.nu assert
assert (1 == ($env.DIRS_LIST | length))
```
3. observe the error

### Expected behavior

```
assert (1 == ($env.DIRS_LIST | length))
```
should work after any `use std.nu ...`

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.77.2                                   |
| branch             | main                                     |
| commit_hash        | ef7fbf4bf9287597175b98082e5273765f828c91 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| build_time         | 2023-03-17 17:50:12 +01:00               |
| build_rust_channel | release                                  |
| features           | default, zip                             |
| installed_plugins  | gstat                                    |

**Comment by fdncred** [comment]
In this instance, this works
```
($base_path | path expand) == $env.DIRS_LIST.0
```
So, maybe we should make sure paths go through `path expand` to ensure they're "right"?

or better yet, it could probably be fixed with this:
```
> $nu.temp-path | path expand
/private/var/folders/m1/x0gp9jy51s146ttxtz7jlpxh0000gn/T
```
versus
```
> $nu.temp-path
/var/folders/m1/x0gp9jy51s146ttxtz7jlpxh0000gn/T/
```

I kind of wonder if that `$nu.temp-path` (and the other paths in $nu) should go through a `path expand` before storing them in nushell's $nu?

**Comment by amtoine** [comment]
> I kind of wonder if that `$nu.temp-path` (and the other paths in $nu) should go through a `path expand` before storing them in nushell's $nu?
>
that might help catch this kind of annoying bug! :yum:

**Comment by amtoine** [comment]
mm wait

the `base_path` is already `expand`ed :thinking: 
https://github.com/nushell/nushell/blob/main/crates/nu-utils/standard_library/test_dirs.nu#L12

**Comment by fdncred** [comment]
> the base_path is already expanded 🤔

ya, i saw that, but it needs it again or it needs to be expanded in the rust code

**Comment by amtoine** [comment]
> ya, i saw that, but it needs it again or it needs to be expanded in the rust code
>
i can try to add more `path expand` and you tell me if that's any better? :thinking:

**Comment by amtoine** [comment]
@fdncred

can you tell me if that's any better with #8552? :yum:

**Comment by fdncred** [comment]
Here are the latest results as of this morning.
```
NU_LOG_LEVEL=DEBUG nu crates/nu-utils/standard_library/tests.nu 
INFO  Run tests in test_asserts
DEBUG Run test test_asserts test_assert
DEBUG Run test test_asserts test_assert_equal
DEBUG Run test test_asserts test_assert_error
DEBUG Run test test_asserts test_assert_greater
DEBUG Run test test_asserts test_assert_greater_or_equal
DEBUG Run test test_asserts test_assert_length
DEBUG Run test test_asserts test_assert_less
DEBUG Run test test_asserts test_assert_less_or_equal
DEBUG Run test test_asserts test_assert_not_equal
INFO  Run tests in test_dirs
DEBUG Run test test_dirs test_dirs_command
Error:
  × add changes PWD to first added dir

INFO  Run tests in test_logger
DEBUG Run test test_logger test_critical
DEBUG Run test test_logger test_debug
DEBUG Run test test_logger test_error
DEBUG Run test test_logger test_info
DEBUG Run test test_logger test_warning
INFO  Run tests in test_std
DEBUG Run test test_std test_match
DEBUG Run test test_std test_path_add
```

**Comment by amtoine** [comment]
the first test output let me say that this issue should not be closed by #8552

**Comment by fdncred** [comment]
> with the first test output i'd say that this issue should not be closed by #8552

it must be something different. i haven't drilled in to see what is going on exactly with line-by-line debugging. but it runs further before puking.


All this makes me glad that we plan to do ci testing cross-platform for std.nu.

**Comment by amtoine** [comment]
> it must be something different. i haven't drilled in to see what is going on exactly with line-by-line debugging. but it runs further before puking.
>
yeah something different than the `base_path` addressed in #8552 :+1: 

but still the issue in this issue is not addressed :open_mouth: 
`std::tests::test_dirs::test_dirs_command` still fails on macOS, right?

worse, maybe we should revert the changes in #8552 if that's not the real problem :thinking: 
we need to investigate more on this :wink: 
this is why i think we should reopen this issue :yum: 

> All this makes me glad that we plan to do ci testing cross-platform for std.nu.
>
oh yeah for sure :relieved:

**Comment by amtoine** [comment]
thanks a lot :yum: 

there is no rush for this, #8525 can wait a bit more :wink:

**Comment by presidento** [comment]
@amtoine 
> you put a 👍 in https://github.com/nushell/nushell/pull/8552, have you tested the fix in macOS to see if the tests run properly without any errors? 😋

No. I did not approve it, I reviewed your code changes and liked it. Sorry if I made confusion with that.

**Comment by presidento** [comment]
> there is no rush for this, https://github.com/nushell/nushell/pull/8525 can wait a bit more 😉

I do not agree. I think it would be better to remove dirs commands, until it is fixed, and wire in the CI without it, since it already breaks the philosophy of being fully cross-platform. (Also it will be easier to test the re-addition in a new PR, since MacOS test will run against it. Until CI is active we cannot be sure that a new command will also break in some platform or not.)

**Comment by amtoine** [comment]
> No. I did not approve it, I reviewed your code changes and liked it. Sorry if I made confusion with that.
>
yeah that's what i thought, do not worry :wink: 
not a big deal at all, things can be reverted and reopened :muscle:

**Comment by amtoine** [comment]
> I do not agree. I think it would be better to remove dirs commands, until it is fixed, and wire in the CI without it, since it already breaks the philosophy of being fully cross-platform. (Also it will be easier to test the re-addition in a new PR, since MacOS test will run against it. Until CI is active we cannot be sure that a new command will also break in some platform or not.)
>
i'd rather disable the tests, e.g. by making theirs names invisible by the test runner, in order to speed up #8525 

then i can solve the merge conflict in the runner and finally merge all this :muscle: 

if everyone is ok with disabling the tests in `test_dirs.nu`, i can do that tomorrow right after work :+1:

**Comment by presidento** [comment]
I do not want to hold it back. If the decision is to disable these tests, please do this way (of course, replace _Windows_ with the proper name for _MacOS_. I do not know, what are the possible values here):

```
diff --git a/crates/nu-utils/standard_library/test_dirs.nu b/crates/nu-utils/standard_library/test_dirs.nu
index 1370c5e35..fe5ec61cf 100644
--- a/crates/nu-utils/standard_library/test_dirs.nu
+++ b/crates/nu-utils/standard_library/test_dirs.nu
@@ -1,5 +1,6 @@
 use std.nu "assert length"
 use std.nu "assert equal"
+use std.nu "log warning"

 def clean [path: path] {
     cd $path
@@ -8,6 +9,10 @@ def clean [path: path] {
 }

 export def test_dirs_command [] {
+    if (sys).host.name == "Windows" {
+        log warning "Skip testing in MacOS, see https://github.com/nushell/nushell/issues/8528"
+        return
+    }
     # need some directories to play with
     let base_path = (($nu.temp-path) | path join $"test_dirs_(random uuid)" | path expand )
     let path_a = ($base_path | path expand | path join "a")
```

**Comment by amtoine** [comment]
@presidento, kinda like the way you disable them :+1:

**Comment by fdncred** [comment]
After talking about this in our meeting today, we do not want to disable any tests, but we want to fix the problems that we find because that just makes the language better. I think we need to fix `path join` so that it expands.

**Comment by fdncred** [comment]
btw #8576 i think fixes the problems we were having. hopefully this will "unstick" us so we can keep moving forward.

**Comment by amtoine** [comment]
i'll try to fix #8525 tonight :+1:
